### PR TITLE
*: add system variable to turn on/off the large transaction feature

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Before the new feature is stable, we must be able to turn on/off it.

### What is changed and how it works?

Add two global/session system variables:

- tidb_enable_large_txn
- tidb_large_txn_size

### Check List 

Tests 

 - Unit test

Code changes

- Has persistent data change

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @coocood @jackysp

**Comment by tiancaiamao** [comment]
PTAL @coocood @jackysp @cfzjywxk

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -920,6 +928,10 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 		s.DisableTxnAutoRetry = TiDBOptOn(val)
 	case TiDBEnableStreaming:
 		s.EnableStreaming = TiDBOptOn(val)
+	case TiDBEnableLargeTxn:
+		s.EnableLargeTxn = TiDBOptOn(val)
+	case TiDBLargeTxnSize:
+		s.LargeTxnSize = tidbOptInt64(val, DefTiDBLargeTxnSize)
```
Comment (cfzjywxk) [inline sessionctx/variable/session.go]: maybe we need to add `TiDBLargeTxnSize`  set value check in `ValidateSetSystemVar`

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -497,19 +501,30 @@ func (c *twoPhaseCommitter) buildPrewriteRequest(batch batchKeys, txnSize uint64
...
 	// When we retry because of a region miss, we don't know the transaction size. We set the transaction size here
 	// to MaxUint64 to avoid unexpected "resolve lock lite".
 	if len(bo.errors) > 0 {
 		txnSize = math.MaxUint64
 	}
 
+	isPrimary := bytes.Equal(batch.keys[0], c.primary())
 	req := c.buildPrewriteRequest(batch, txnSize)
+
+	if x := bo.ctx.Value("checkLargeTxnEnable"); x != nil {
```
Comment (coocood) [inline store/tikv/2pc.go]: This has some extra cost, can we avoid it?

File: sessionctx/variable/sysvar.go [hunk 3]
```diff
@@ -702,6 +702,8 @@ var defaultSysVars = []*SysVar{
 	{ScopeGlobal | ScopeSession, TiDBEnableVectorizedExpression, BoolToIntStr(DefEnableVectorizedExpression)},
 	{ScopeGlobal | ScopeSession, TiDBEnableFastAnalyze, BoolToIntStr(DefTiDBUseFastAnalyze)},
 	{ScopeGlobal | ScopeSession, TiDBSkipIsolationLevelCheck, BoolToIntStr(DefTiDBSkipIsolationLevelCheck)},
+	{ScopeGlobal | ScopeSession, TiDBEnableLargeTxn, BoolToIntStr(DefTiDBEnableLargeTxn)},
+	{ScopeGlobal | ScopeSession, TiDBLargeTxnSize, strconv.Itoa(DefTiDBLargeTxnSize)},
```
Comment (coocood) [inline sessionctx/variable/sysvar.go]: I think we can use a constant for it instead of a session variable.

File: sessionctx/variable/sysvar.go [hunk 4]
```diff
@@ -702,6 +702,8 @@ var defaultSysVars = []*SysVar{
 	{ScopeGlobal | ScopeSession, TiDBEnableVectorizedExpression, BoolToIntStr(DefEnableVectorizedExpression)},
 	{ScopeGlobal | ScopeSession, TiDBEnableFastAnalyze, BoolToIntStr(DefTiDBUseFastAnalyze)},
 	{ScopeGlobal | ScopeSession, TiDBSkipIsolationLevelCheck, BoolToIntStr(DefTiDBSkipIsolationLevelCheck)},
+	{ScopeGlobal | ScopeSession, TiDBEnableLargeTxn, BoolToIntStr(DefTiDBEnableLargeTxn)},
```
Comment (coocood) [inline sessionctx/variable/sysvar.go]: I think we can use a config file option, so we can deprecate it later.

File: store/tikv/2pc.go (hunk 2 above)
Comment (tiancaiamao) [reply store/tikv/2pc.go]: I tried failpoint but the concurrent test is a problem.

File: sessionctx/variable/sysvar.go (hunk 3 above)
Comment (tiancaiamao) [reply sessionctx/variable/sysvar.go]: Constant for `TiDBEnableLargeTxn` and `TiDBLargeTxnSize` ? or for  `DefTiDBXXX`

File: store/tikv/2pc.go (hunk 2 above)
Comment (coocood) [reply store/tikv/2pc.go]: The test doesn't make much sense.

**Comment by tiancaiamao** [comment]
@coocood disagree with adding session variables to turn on this new feature.

**Pull request closed without merging**